library/ns1_zone.py

```diff
@@ -489,19 +489,15 @@ def diff_in_secondaries(self, have_secondaries, want_secondaries):
         return False
 
     def convert_secondaries_to_dict(self, secondaries):
-        """Converts a secondaries list to a dictionary. Keys are a concatenated string
+        """Converts a secondaries list to a dictionary. Keys are a tuple of
         of IP and Port fields.
 
         :param secondaries: List of secondary dicts
         :type secondaries: list
         :return: Dict of secondary dicts
         :rtype: dict
         """
-        secondaries_dict = {}
-        for secondary in secondaries:
-            socket = "{0}:{1}".format(secondary["ip"], secondary["port"])
-            secondaries_dict[socket] = secondary
-        return secondaries_dict
+        return {(s.get("ip"), s.get("port")): s for s in secondaries}
```

Comment:
dig the comprehension, but are you sure the `.get`s are OK? Previous impl would throw KeyError if anything is awry, this one will continue (and might clobber values due to e.g. `(None, None)` keys.

Reply:
True.  Was trying to be defensive against `KeyError`, because effectively missing key _is_ a unique entry.  On the other hand clobbering vals with the `(None, None)` corner case is a possibility, and in the happy path these should never be missing, so catching `KeyError` and throwing an Ansible error is prob better solution.